core/hoverfly_funcs_test.go

```diff
@@ -336,6 +341,7 @@ func Test_Hoverfly_GetResponse_WillCacheTemplateIfNotInCache(t *testing.T) {
 	Expect(found).To(BeTrue())
 
 	Expect(cachedRequestResponsePair.(*models.CachedResponse).MatchingPair.Response.Body).To(Equal("{{ randomUuid }}"))
+	Expect(cachedRequestResponsePair.(*models.CachedResponse).MatchingPair.Response.Headers["X-Image-Id"][0]).To(Equal("{{ randomInteger }}"))
```

Comment:
Perhaps you can try adding a test in this test suite to check if the header template is rendered properly? Or you can add a functional test here: https://github.com/SpectoLabs/hoverfly/blob/master/functional-tests/core/ft_templated_response_test.go
Whichever you prefer.

Reply:
Add unit-test `core.Test_Hoverfly_GetResponse_WillCacheHeaderTemplateIfNotInCache` and functional-test `randomInteger in header` in file `functional-tests/core/ft_templated_response_test.go`